openmw: 0.46 -> 0.47

marius851000 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

OpenMW version 0.47 was released, with some really interesting new feature (was using a nightly version compiled via Nix previously, on which this commit is based)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).


note : 
openmw dropped support for Qt4 (so removed the compilation flag)
it add two dependancies, a direct one on recastnavigation, and an indirect on OSG with collada (adding some support for collada format)
The optional opencollada depencie of OSG was broken. I repeared it, using the code referenced in the OpenMW CI.
Added myself to OpenMW maintainers.

marius851000 committed:
recastnavigation: init at 2021-03-05

marius851000 committed:
collada-dom: init at rdiankov-2020-01-03

marius851000 committed:
openscenegraph: fix collada support

marius851000 commented:
openmw maintainer : @abbradar 
openscenegraph maintainer (fixing collada support) : @aanderse @7c6f434c

7c6f434c reviewed (approved):
Looks fine (not that I know what Collada does, but the changes seem to make sense)

aanderse reviewed (approved):
LGTM :+1: Thanks!

marius851000 commented:
Updated the OpenMW bullet to fix physics mulithreading. I will see if I should update the global bullet package, but that will for another PR.

thiagokokada commented on pkgs/development/libraries/collada-dom/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+stdenv.mkDerivation {
+  pname = "collada-dom";
+  version = "rdiankov-2020-01-03";
```
```suggestion
  version = "unstable-2020-01-03";
```

thiagokokada commented on pkgs/development/libraries/recastnavigation/default.nix:
```diff
@@ -0,0 +1,27 @@
+  src = fetchFromGitHub {
+    owner = "recastnavigation";
+    repo = "recastnavigation";
+    rev = "c5cbd53024c8a9d8d097a4371215e3342d2fdc87";
```
Checks: https://github.com/recastnavigation/recastnavigation/commit/c5cbd53024c8a9d8d097a4371215e3342d2fdc87

thiagokokada commented on pkgs/development/libraries/recastnavigation/default.nix:
```diff
@@ -0,0 +1,27 @@
+  pname = "recastai";
+  # use latest revision for the CMake build process and OpenMW
+  # OpenMW use e75adf86f91eb3082220085e42dda62679f9a3ea
+  version = "2021-03-05";
```
```suggestion
  version = "unstable-2021-03-05";
```

thiagokokada commented on pkgs/games/openmw/default.nix:
```diff
@@ -13,56 +13,75 @@
 
 let
-  openscenegraph_ = openscenegraph.overrideDerivation (self: {
+  openscenegraph_openmw = (openscenegraph.override {colladaSupport = true; })
```
```suggestion
  openscenegraph_openmw = (openscenegraph.override { colladaSupport = true; })
```

thiagokokada reviewed (commented):
Almost good, just some fixes.

marius851000 commented on pkgs/development/libraries/collada-dom/default.nix:
```diff
@@ -0,0 +1,35 @@
+  src = fetchFromGitHub {
+    owner = "rdiankov";
+    repo = "collada-dom";
+    rev = "c1e20b7d6ff806237030fe82f126cb86d661f063";
```
check what ? I don't understand.

thiagokokada replied:
It is just that I checked this commit matches a commit from the repository.

thiagokokada commented:
Result of `nixpkgs-review pr 144798` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>openmw-tes3mp</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>collada-dom</li>
    <li>openmw</li>
    <li>recastnavigation</li>
  </ul>
</details>

thiagokokada commented:
Is `openmw-tes3mp` failure something that is supposed to happen?

1MachineElf commented:
This OpenMW 0.47 nix package seems to work for me. :)

marius851000 commented:
Oops (for tes3mp not compiling). Here is what happen I forget to run nixpkgs-review... Will take a look at it back at home.

SuperSandro2000 commented on pkgs/games/openmw/default.nix:
```diff
@@ -13,56 +13,75 @@
 
   meta = with lib; {
     description = "An unofficial open source engine reimplementation of the game Morrowind";
     homepage = "http://openmw.org";
```
```suggestion
    homepage = "https://openmw.org/";
```

SuperSandro2000 commented on pkgs/games/openmw/default.nix:
```diff
@@ -13,56 +13,75 @@
 in
 mkDerivation rec {
-  version = "0.46.0";
+  version = "0.47.0";
   pname = "openmw";
```
```suggestion
  pname = "openmw";
  version = "0.47.0";
```

SuperSandro2000 commented on pkgs/games/openmw/default.nix:
```diff
@@ -13,56 +13,75 @@
+    });
+
+  bullet_openmw = bullet.overrideDerivation (old: rec {
+    version = "3.17";
```
Can we update bullet or add buller_3 instead?

marius851000 replied:
I don't know. I'll probably take a look at that later, but seeing there are other software that depend on bullet, and that it seems to be a major new version, I plan to make another PR dedicated to this.

marius851000 replied:
Well, it seems updating the library doesn't cause any build failure nor cause runtime failure on vdrift or stuntrally. I'll open a different PR for it,

marius851000 replied:
Done here : https://github.com/NixOS/nixpkgs/pull/145385

SuperSandro2000 commented on pkgs/development/libraries/collada-dom/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  postInstall = ''
+    chmod +w -R $out
+    mv $out/include/*/* $out/include
```
Maybe we should symlink to not break imports of other packages?

SuperSandro2000 commented on pkgs/development/libraries/collada-dom/default.nix:
```diff
@@ -0,0 +1,35 @@
+    owner = "rdiankov";
+    repo = "collada-dom";
+    rev = "c1e20b7d6ff806237030fe82f126cb86d661f063";
+    sha256 = "A1ne/D6S0shwCzb9spd1MoSt/238HWA8dvgd+DC9cXc=";
```
```suggestion
    sha256 = "sha256-A1ne/D6S0shwCzb9spd1MoSt/238HWA8dvgd+DC9cXc=";
```

SuperSandro2000 commented:
thanks @thiagokokada and the others. PR is already in a really good state!

marius851000 committed:
openmw: 0.46 -> 0.47

marius851000 commented:
So, it seems that my checks (https://github.com/NixOS/nixpkgs/pull/145385) confirm that tes3mp compiled on master...

thiagokokada commented:
> So, it seems that my checks (#145385) confirm that tes3mp compiled on master...

Just to understand here, we should merge #145385 and afterwards this one?

marius851000 commented:
Oh. I just meant that it fetched tes3mp from the hydra cache and that it compiled, so I should broken it somehow here (it's compiling right now on this branch)

marius851000 commented:
So, I fixed tes3mp compilation. But I have issue going on a public server (due to a version mismatch). I'll have to fix this later (but doesn't seems worse than the current version). Also added a small patch that would allow ``tes3mp-browser`` to correctly locate the binary (instead of using the one in the current folder).

thiagokokada commented:
> So, I fixed tes3mp compilation. But I have issue going on a public server (due to a version mismatch). I'll have to fix this later (but doesn't seems worse than the current version). Also added a small patch that would allow `tes3mp-browser` to correctly locate the binary (instead of using the one in the current folder).

Do you want to fix this in this PR? Or can I go and merge it as-is (since it isn't worse than it current status)?

marius851000 commented:
I'll probably take a look later, and I'll prefer to make another PR for fixing the tes3mp version.

thiagokokada commented:
Result of `nixpkgs-review pr 144798` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>collada-dom</li>
    <li>openmw</li>
    <li>openmw-tes3mp</li>
    <li>portmod</li>
    <li>recastnavigation</li>
  </ul>
</details>

thiagokokada commented:
Thanks folks. Merged.

Future improvements can come in other PRs.
